### ISSUE TITLE
Tensorrt output NaN while onnxruntime is fine

### ISSUE THREAD

**Issue by tpoisonooo** [NVIDIA/TensorRT#2338]
## Description

Use `polygraphy run --trt` and got NaN output.

## Environment

**TensorRT Version**:  8.4.1.5
**NVIDIA GPU**:  NVIDIA GeForce GTX 1660 SUPER
**NVIDIA Driver Version**: 510.85.02
**CUDA Version**: 11.6
**CUDNN Version**: 8.4 (also failed on cudnn8.2)
**Operating System**: Ubuntu18.04
**Python Version (if applicable)**: 3.7.4

## Relevant Files

<!-- Please include links to any models, data, files, or scripts necessary to reproduce your issue. (Github repo, Google Drive/Dropbox, etc.) -->

Please download `pointpillars_trt.tar` from  https://drive.google.com/file/d/1FuZJWLIsJyUsUk_lM1euXzyPgagu-tXj/view?usp=sharing  and untar.

```bash
$ tree
├── coors.npy
├── custom_inputs.json
├── data_loader.py
├── end2end.onnx
├── num_points.npy
└── voxels.npy
```

## Steps To Reproduce

<!-- 
  Craft a minimal bug report following this guide - https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports

  Please include:
  * Exact steps/commands to build your repro
  * Exact steps/commands to run your repro
  * Full traceback of errors encountered 
-->

1. install polygraphy
```bash
$ polygraphy --version
Polygraphy | Version: 0.38.0 | Path: ['/home/user/anaconda3/envs/repo2.0/lib/python3.7/site-packages/polygraphy']
```

2. run `polygraphy` and feed custom data  and got NaN
```bash
$ polygraphy run end2end.onnx  --plugins  --trt --data-loader-script data_loader.py \
 --trt-min-shapes voxels:[2000,32,4]  num_points:[2000,] coors:[2000,4] \
--trt-opt-shapes voxels:[3000,32,4] num_points:[3000,] coors:[3000,4] \
--trt-max-shapes voxels:[9000,32,4] num_points:[9000,] coors:[9000,4] \
 -vv  --validate  --workspace 4096M

..

[E]         Inf Detected | One or more non-finite values were encountered in this output                                                                                                        
[X]         Note: non-finite values at:                                                                                                                                                         
            [[[[ True  True  True ... False False False]                                                                                                                                        
               [ True  True  True ... False False False]                                                                                                                                        
               [ True  True  True ... False False False]                                                                                                                                        
               ...                                                                                                                                                                              
               [False False False ... False False False]                                                                                                                                        
               [False False False ... False False False]                                                                                                                                        
               [False False False ... False False False]]                                                                                                                                       
                                                                                                                                                                                                
              [[ True  True  True ... False False False]                                                                                                                                        
               [ True  True  True ... False False False]                                                                                                                                        
               [ True  True  True ... False False False]                                                                                                                                        
               ...                                                                                                                                                                              
               [False False False ... False False False]                                                                                                                                        
               [False False False ... False False False]                                                                                                                                        
               [False False False ... False False False]]                       
```

3. For `--onnxrt` as backend, everything would be fine.

4. Notes
A. This bug **also** existed in [jetson orin 32G with jetpack5.0.2](https://github.com/open-mmlab/mmdeploy/issues/937)
B. This bug **cannot** reproduce on ubuntu18+x86_64+cu102+trt 8.4.1.5

**Comment by zerollzeng** [comment]
I can reproduce this in TRT 8.5.0.9. but the issue is gone when I don't use dynamic shape.
```
[I] trt-runner-N0-09/22/22-00:17:14     | Completed 1 iteration(s) in 17.49 ms | Average inference time: 17.49 ms.
[I] Output Validation | Runners: ['trt-runner-N0-09/22/22-00:17:14']
[I]     trt-runner-N0-09/22/22-00:17:14     | Validating output: cls_score (check_inf=True, check_nan=True)
[I]         mean=-9.7008, std-dev=2.5247, var=6.3741, median=-9.8573, min=-19.325 at (0, 10, 233, 29), max=2.9861 at (0, 14, 120, 46), avg-magnitude=9.701
[I]         PASSED | Output: cls_score is valid
[I]     trt-runner-N0-09/22/22-00:17:14     | Validating output: bbox_pred (check_inf=True, check_nan=True)
[I]         mean=-0.057552, std-dev=0.47462, var=0.22527, median=0.0027663, min=-4.9191 at (0, 13, 129, 13), max=2.5795 at (0, 6, 92, 56), avg-magnitude=0.18281
[I]         PASSED | Output: bbox_pred is valid
[I]     trt-runner-N0-09/22/22-00:17:14     | Validating output: dir_cls_pred (check_inf=True, check_nan=True)
[I]         mean=2.0734e-05, std-dev=0.72226, var=0.52167, median=-0.00012761, min=-7.6589 at (0, 5, 119, 42), max=7.659 at (0, 4, 119, 42), avg-magnitude=0.50724
[I]         PASSED | Output: dir_cls_pred is valid
[I]     PASSED | Output Validation
[I] PASSED | Command: /home/zeroz/.local/bin/polygraphy run end2end_foled.onnx --trt --data-loader-script data_loader.py --trt-min-shapes voxels:[3945,32,4] num_points:[3945,] coors:[3945,4] --trt-opt-shapes voxels:[3945,32,4] num_points:[3945,] coors:[3945,4] --trt-max-shapes voxels:[3945,32,4] num_points:[3945,] coors:[3945,4] --validate

```

**Comment by zerollzeng** [comment]
I guess the reason is related to some cuda kernel under cuda 11.6. will investigate further

**Comment by tpoisonooo** [comment]
> 

This model is centerpoint which is used in Automatic Drive, input shape must be dynamic ..

**Comment by zerollzeng** [comment]
This issue is fixed in 8.5.1 with FASTER_DYNAMIC_SHAPES_0805 preview feature. it should come with our TRT 8.5 release, 
```
[I]     PASSED | Output Validation
[V] Loaded Module: sys
[I] PASSED | Command: /home/zeroz/.local/bin/polygraphy run end2end_foled.onnx --trt --data-loader-script data_loader.py --trt-min-shapes voxels:[2000,32,4] num_points:[2000,] coors:[2000,4] --trt-opt-shapes voxels:[3000,32,4] num_points:[3000,] coors:[3000,4] --trt-max-shapes voxels:[9000,32,4] num_points:[9000,] coors:[9000,4] -vv --validate --preview-features FASTER_DYNAMIC_SHAPES_0805
```
cc @jackwish @pranavm-nvidia
